fatiando/mesher.py

```diff
@@ -1721,3 +1721,594 @@ def keep(cell):
+                                                              self.dip,
+                                                              self.rake)
+
+        # Coordinate transformation matrix
+        self.transf_matrix = coord_transf_matrix_oblate(self.alpha,
+                                                        self.gamma,
+                                                        self.delta)
+
+    def __str__(self):
+        """
+        Return a string representation of the oblate ellipsoid.
+        """
+
+        names = [('x', self.x), ('y', self.y), ('z', self.z),
+                 ('small_axis', self.small_axis),
+                 ('large_axis', self.large_axis), ('strike', self.strike),
+                 ('dip', self.dip), ('rake', self.rake)]
+        names = names + [(p, self.props[p]) for p in sorted(self.props)]
+        return ' | '.join('%s:%s' % (n, v) for n, v in names)
+
+    def susceptibility_tensor(self):
+        '''
+        Calculate the susceptibility tensor (in SI) in the main system.
+
+        The susceptibility tensor is calculated if 'susceptibility tensor'
+        is defined in the dictionary of physical properties props.
+        In this case, 'susceptibility tensor' must have six elements:
+        three positive eigenvalues (principal susceptibilities), in
+        descending order, and three angles (in degrees) defining
+        the eigenvector matrix U of the susceptibility tensor, respectively.
+        The eigenvector matrix is defined by the function
+        coord_transf_matrix_oblate.
+        '''
+
+        assert 'susceptibility tensor' in self.props, 'susceptibility tensor \
+is not in the dictionary of physical properties'
+
+        assert len(self.props['susceptibility tensor']) == 6, 'susceptibili\
+ty tensor must be a list containing six elements'
+
+        # Large, intermediate and small eigenvalues of the
+        # susceptibility tensor (principal susceptibilities)
+        k1 = self.props['susceptibility tensor'][0]
+        k2 = self.props['susceptibility tensor'][1]
+        k3 = self.props['susceptibility tensor'][2]
+
+        assert k1 >= k2 >= k3, 'the eigenvalues must be given in \
+descending order'
+
+        assert (k1 > 0) and (k2 > 0) and (k3 > 0), 'the eigenvalues must \
+be all positive'
+
+        # Angles (in degrees) defining the eigenvector matrix
+        # of the susceptibility tensor
+        strike = self.props['susceptibility tensor'][3]
+        dip = self.props['susceptibility tensor'][4]
+        rake = self.props['susceptibility tensor'][5]
+
+        # Eigenvector matrix of the susceptibility tensor
+        alpha, gamma, delta = auxiliary_angles(strike, dip, rake)
+        U = coord_transf_matrix_oblate(alpha, gamma, delta)
+
+        suscep_tensor = numpy.dot(U, numpy.diag([k1, k2, k3]))
+        suscep_tensor = numpy.dot(suscep_tensor, U.T)
+
+        return suscep_tensor
+
+
+def auxiliary_angles(strike, dip, rake):
```

Comment:
This method is very specific to the ellipsoid code so it should be private (add a `_` to the beginning of the name).